### PR TITLE
Hide pending

### PR DESCRIPTION
**Pull request by johnlucia**
Service needs to hide corporate locations from results.

Also needs to hide Deleted or Suspended locations.

Pending locations should only show up if the corporate location is also Pending

### CODE REVIEW

File: app/models/client.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Client < G5Updatable::Client
```
Comment (variousred) [inline app/models/client.rb]: for now (until we converge on a common solution for this nastiness) I'd do this the way some of the other apps do e.g.: https://github.com/g5search/g5-content-management-system/blob/95194410c58da32bd411d5609412859c402f77ee/app/services/bootstrap_g5_updatable_client.rb

File: app/controllers/clients_controller.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@ def index
   end
 
   def location_search
-    @client = G5Updatable::Client.find_by_urn!(params[:client_id])
+    @client = Client.find_by_urn!(params[:client_id])
```
Comment (variousred) [inline app/controllers/clients_controller.rb]: ```The Rails 2 style dynamic finder methods are finally being deprecated as of Rails 4. The code has been extracted to a gem named activerecord-deprecated_finders. The gem will be a dependency in Rails 4.0, but will be removed as of Rails 4.1.``` not gone yet, but deprecated — new syntax is ```find_by(urn: params[:client_id])```

File: lib/radius_search.rb [hunk 3]
```diff
@@ -2,7 +2,8 @@ class RadiusSearch
   include ApplicationHelper
 
   def initialize(client, params)
-    @client_locations = Location.where({client_uid: client.uid, corporate: false})
+    statuses = client.is_pending ? ["Live", "Pending", "No Deploy"] : ["Live", "No Deploy"]
+    @client_locations = Location.where({client_uid: client.uid, corporate: false, status: statuses})
```
Comment (variousred) [inline lib/radius_search.rb]: I could see this being useful as a scope on ```Location```

File: lib/radius_search.rb [hunk 4]
```diff
@@ -11,7 +12,7 @@ def locations
   end
 
   def results
-    success = self.locations.empty? ? false : true
+    success = self.locations.any?
```
Comment (levibrown) [inline lib/radius_search.rb]: nice

File: app/models/client.rb (hunk 1 above)
Comment (variousred) [inline app/models/client.rb]: re: conversation in slack, i really like this actually. do associations still work though is a question that came up, especially the polymorphic one of website — owner, where the owner type would be ```G5Updatable::Client```

Comment (variousred) [inline app/models/client.rb]: nm this is not CMS ...

**Pull request merged**